src/main/java/com/linecorp/armeria/client/circuitbreaker/EventCounter.java

```diff
@@ -25,12 +27,18 @@
 
     /**
      * Counts success event
+     *
+     * @return An {@link Optional} containing the current {@link EventCount} if it has been updated, or else
+     * an empty {@link Optional}.
      */
-    void onSuccess();
+    Optional<EventCount> onSuccess();
```

Comment:
In rough estimation, This method creates a new `Optional` instance every `counterUpdateInterval`  (1 second in default). Otherwise it returns `empty`(singleton). In that case, I think it isn't a problem. But it might be a problem indeed when user sets a very short interval...